Resync Secure Contexts WPT

rreno opened the pull request:
#### 7c9d4354393a7ef4aeb49f0cc6300b78acf889e6
<pre>
Resync Secure Contexts WPT
<a href="https://bugs.webkit.org/show_bug.cgi?id=249335">https://bugs.webkit.org/show_bug.cgi?id=249335</a>
rdar://103371925

Reviewed by Brent Fulgham.

Resync Secure Contexts WPT as of upstream commit 8adf127d65cbca5895a4e665dcb9e763e79abc79

* LayoutTests/TestExpectations:
* LayoutTests/platform/mac-wk1/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/secure-contexts/META.yml: Added.
* LayoutTests/imported/w3c/web-platform-tests/secure-contexts/basic-dedicated-worker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/secure-contexts/basic-dedicated-worker.html:
* LayoutTests/imported/w3c/web-platform-tests/secure-contexts/basic-dedicated-worker.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/secure-contexts/basic-popup-and-iframe-tests-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/secure-contexts/basic-popup-and-iframe-tests.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/secure-contexts/basic-popup-and-iframe-tests.https.js:
* LayoutTests/imported/w3c/web-platform-tests/secure-contexts/basic-shared-worker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/secure-contexts/basic-shared-worker.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/secure-contexts/shared-worker-insecure-first.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/secure-contexts/shared-worker-secure-first.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/secure-contexts/support/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/secure-contexts/w3c-import.log:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/secure-contexts/basic-dedicated-worker-expected.txt: Added.
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/secure-contexts/basic-popup-and-iframe-tests-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/secure-contexts/basic-popup-and-iframe-tests.https-expected.txt: Copied from LayoutTests/platform/glib/imported/w3c/web-platform-tests/secure-contexts/basic-popup-and-iframe-tests-expected.txt.

Canonical link: <a href="https://commits.webkit.org/257994@main">https://commits.webkit.org/257994@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6f92b8ebafe4f1f61c3633faba91373506ac6a69

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/100455 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/9614 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/33516 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/109789 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/170044 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/10526 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/194 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/92923 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/107642 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/106234 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/7964 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/91232 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/34632 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/89880 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/22621 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/77599 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/3349 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/24140 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/3342 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/284 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/9462 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/43630 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5480 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/5156 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on LayoutTests/imported/w3c/web-platform-tests/secure-contexts/basic-shared-worker-expected.txt:
```diff
@@ -0,0 +1,10 @@
+Blocked access to external URL https://www.localhost:9443/secure-contexts/support/https-subframe-shared.html
+
+
+Harness Error (TIMEOUT), message = null
```
We might want to SKIP this test due to timeout.

brentfulgham commented on LayoutTests/TestExpectations:
```diff
@@ -3049,10 +3049,6 @@ fast/table/colspanMinWidth-vertical.html [ Skip ]
-# Secure Contexts tests that timeout
-imported/w3c/web-platform-tests/secure-contexts/basic-dedicated-worker.html [ Skip ]
-imported/w3c/web-platform-tests/secure-contexts/basic-dedicated-worker.https.html [ Skip ]
-imported/w3c/web-platform-tests/secure-contexts/basic-popup-and-iframe-tests.https.html [ Skip ]
```
Should we add:

imported/w3c/web-platform-tests/secure-contexts/basic-shared-worker-expected.txt [ Skip ]
imported/w3c/web-platform-tests/secure-contexts/basic-shared-worker.https-expected.txt [ Skip ]
imported/w3c/web-platform-tests/secure-contexts/shared-worker-insecure-first.https-expected.txt [ Skip ]
imported/w3c/web-platform-tests/secure-contexts/shared-worker-secure-first.https-expected.txt [ Skip ]

rreno replied:
Timeouts are already skipped. See https://github.com/WebKit/WebKit/blob/main/LayoutTests/TestExpectations#L305

brentfulgham reviewed (approved):
LGTM, but please skip those timeouts.

rreno committed:
Resync Secure Contexts WPT
https://bugs.webkit.org/show_bug.cgi?id=249335
rdar://103371925

Reviewed by Brent Fulgham.

Resync Secure Contexts WPT as of upstream commit 8adf127d65cbca5895a4e665dcb9e763e79abc79

* LayoutTests/TestExpectations:
* LayoutTests/platform/mac-wk1/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/secure-contexts/META.yml: Added.
* LayoutTests/imported/w3c/web-platform-tests/secure-contexts/basic-dedicated-worker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/secure-contexts/basic-dedicated-worker.html:
* LayoutTests/imported/w3c/web-platform-tests/secure-contexts/basic-dedicated-worker.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/secure-contexts/basic-popup-and-iframe-tests-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/secure-contexts/basic-popup-and-iframe-tests.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/secure-contexts/basic-popup-and-iframe-tests.https.js:
* LayoutTests/imported/w3c/web-platform-tests/secure-contexts/basic-shared-worker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/secure-contexts/basic-shared-worker.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/secure-contexts/shared-worker-insecure-first.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/secure-contexts/shared-worker-secure-first.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/secure-contexts/support/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/secure-contexts/w3c-import.log:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/secure-contexts/basic-dedicated-worker-expected.txt: Added.
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/secure-contexts/basic-popup-and-iframe-tests-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/secure-contexts/basic-popup-and-iframe-tests.https-expected.txt: Copied from LayoutTests/platform/glib/imported/w3c/web-platform-tests/secure-contexts/basic-popup-and-iframe-tests-expected.txt.

Canonical link: https://commits.webkit.org/257994@main
